### PR TITLE
Migrate `test-benches`, `c-unwind-abi-catch-panic` and `compiler-lookup-paths-2` `run-make` tests to rmake

### PR DESCRIPTION
**Pull request by Oneirical**
Part of #121876 and the associated [Google Summer of Code project](https://blog.rust-lang.org/2024/05/01/gsoc-2024-selected-projects.html).

### CODE REVIEW

File: tests/run-make/test-benches/rmake.rs [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+//@ ignore-cross-compile
+// Reason: the compiled binary is executed
+//@ needs-unwind
+// Reason: #[bench] and -Zpanic-abort-tests can't be combined
+
+use run_make_support::{run, run_with_args, rustc};
+
+fn main() {
+    rustc().arg("--test").input("smokebench.rs").opt().run();
```
Comment (Kobzol) [inline tests/run-make/test-benches/rmake.rs:16]: `// Smoke-test that `#[bench]` isn't entirely broken.`
I think we could keep the original comment here.

**Review by Kobzol — COMMENTED** [review]
Thanks! Left one nit, but I don't think it's super important. Feel free to r=me with or without fixing it.

r? @Kobzol

**Comment by Kobzol** [comment]
@bors delegate+

(sigh)

**Comment by Oneirical** [comment]
@bors r=@Kobzol

Thanks for the review!

**Pull request merged**